File: Laba2/Task1.cs
using Extensions;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Serialization;

namespace PZPK_Labas.Laba2
{
    public static class Task1
    {
        public static void Run()
        {
            Console.WriteLine("Завдання 1. " + "Дано ціле-чисельний масив, що складається із n (позитивних та негативних) елементів.\n" + 
                "Отримати новий масив, як різницю між елементами вхідного масиву та його середнього арифметичного.");

            int n;
            ConsoleInput.SafeIntInput("Введіть кількість елементів масиву", out n);

            int[] array = ArrayHelper.GenerateArray(n, 0, 20);

            Console.WriteLine("\nПочатковий масив:");
            ArrayHelper.PrintArray(array);

            Console.WriteLine($"\nСереднє арифметичне масиву: {GetAveregeNumberOfArray(array)}\n");

            Console.WriteLine("Оброблений масив:");
            ArrayHelper.PrintArray(GetModifiedArray(array, GetAveregeNumberOfArray(array)));

            Console.WriteLine();
            Console.ReadLine();
        }

        private static int[] GetModifiedArray(int[] array, int average)
        {
            int[] newArray = new int[array.Length];

            for (int i = 0; i < array.Length; i++)
                newArray[i] = array[i] - average;

            return newArray;
        }

        private static int GetAveregeNumberOfArray(int[] array)
        {
            int sum = 0;

            for (int i = 0; i < array.Length; i++)
                sum += array[i];

            return sum / array.Length;
        }
    }
}

File: Laba2-Python/Task1.py
import Extensions
import ArrayHelper
import sys

def Run():
    print("Завдання 1. " + "Дано ціле-чисельний масив, що складається із n (позитивних та негативних) елементів.\n" + 
        "Отримати новий масив, як різницю між елементами вхідного масиву та його середнього арифметичного.")

    n = 0
    try:
        n = int(input("Введіть кількість елементів масиву: "))
    except ValueError:
        print("Некоректне значення.")
        sys.exit()

    array = ArrayHelper.GenerateArray(n, 0, 20)

    print("\nПочатковий масив:")
    ArrayHelper.PrintArray(array)

    average = GetAveregeNumberOfArray(array)
    print(f"\nСереднє арифметичне масиву: {average}\n")

    print("Оброблений масив:")
    modifiedArray = GetModifiedArray(array, average)
    ArrayHelper.PrintArray(modifiedArray)

    print()
    input()

def GetModifiedArray(array, average):
    newArray = [0] * len(array)

    for i in range(len(array)):
        newArray[i] = array[i] - average

    return newArray

def GetAveregeNumberOfArray(array):
    sum = 0

    for number in array:
        sum += number

    return sum // len(array)